cep-recipe-serialization.md

```diff
@@ -0,0 +1,1110 @@
+  spec: some-pkg-a >=1.0.0,<2
+- source: run_export
+  spec: libzlib >=1.3.1,<1.4.0a0
+  from: host # environment that exported this dependency
+  source_package: zlib # source package that exported this dependency
+- source: variant
+  spec: setuptools 69.*
+  variant: "69" # variant coming from the variant configuration
+```
+
+The list of `resolved` dependencies contains a dictionary with the following
+fields (equivalent to `RepoDataRecord`):
+
+```yaml
+build: hb7217d7_0
+build_number: 0
+depends: []
+license: GPL-2.0-or-later
+license_family: GPL
+md5: fe8efc3385f58f0055e8100b07225055
+name: libtool
+sha256: efe50471d2baea151f2a93f1f99c05553f8c88e3f0065cdc267e1aa2e8c42449
+size: 408403
+subdir: osx-arm64
+timestamp: 1672362003513 # when available
+version: 2.4.7
+fn: libtool-2.4.7-hb7217d7_0.conda
+url: https://conda.anaconda.org/conda-forge/osx-arm64/libtool-2.4.7-hb7217d7_0.conda
+channel: https://conda.anaconda.org/conda-forge/
+```
+
+#### Run exports
+
+The `build` and `host` section contain a `run_exports` dictionary that contains
+collected run_exports from all direct dependencies. The dictionary contains
+`name: run_export` pairs. A `run_export` entry contains the keys:
+
+```yaml
+weak: [list of weak run exports]
+strong: [list of strong run exports]
+weak_constrains: [list of weak constrains]
+strong_constrains: [list of strong constrains]
+noarch: [list of noarch run exports]
+```
+
+The `run` section also contains a single `run_exports` entry that is used to
+generate the `run_exports` metadata for the package that we are currently
+building. Different from the other `run_exports`, these are "outgoing"
+run_exports.
+
+### Finalized sources
+
+This section contains the finalized sources that were used at build-time. For
+`git` sources, this contains a reference to the commit hash that was used. For
+`url` and `path` sources, this contains the URL and the SHA256 hash of the
+downloaded file.
+
+```yaml
+finalized_sources:
+  - path: ../
+  - git: https://github.com/ros2-gbp/rosidl_typesupport_fastrtps-release.git
+    rev: 7af149d83d7e4a2d4c9ce4ae9682bff4a8ebc28e # here we are storing the commit hash
+    target_directory: ros-humble-fastrtps-cmake-module/src/work
+  - url: https://github.com/watchexec/watchexec/archive/refs/tags/v1.25.1.tar.gz
+    sha256: 9609163c14cd49ec651562838f38b88ed2d370e354af312ddc78c2be76c08d37
+```
+
+The following fields are available for each source:
+
+#### Path source
+
+> [!NOTE] The `sha256` field is added if it is not already present in the source
+> description. (TODO in rattler-build) We should compute the SHA hashes of the
+> patches as well and store it in the source description.
+
+```yaml
+path: # path to the source
+sha256: # computed sha256 hash of the source (this is added if not in the source description)
+patches: # list of patches that should be applied to the source
+target_directory: # optional target directory where the source should be extracted
+file_name: # optional file name
```

Comment:
Wouldn't the filename be part of `path`?

Reply:
Filename is more for renaming the file in the work directory. So yes, you could add a `file_name` argument if you want (and then it would be rendered).